### ISSUE TITLE
🐞 Bug: `markview.state.detach()` is called when vim.bo.buflisted=false

### ISSUE THREAD

**Issue by ksjun** [OXY2DEV/markview.nvim#463]
### 🤖 Which branch are you using?

- [ ] dev
- [x] main

### 📜 Issue description:

There seems to be no reason to call detach() since the buffer is still alive, it just disappeared from the buffer list.

https://github.com/OXY2DEV/markview.nvim/blob/18fc148c83925919d94f3fc8e5c74f2d108b9b44/lua/markview/autocmds.lua#L433

### 📑 Reproduction steps:

1. load Markdown file.
2. ```:lua =require('markview.state').buf_attached(vim.api.nvim_get_current_buf())```
3. ```:lua vim.bo.buflisted=false```
4. ```:lua =require('markview.state').buf_attached(vim.api.nvim_get_current_buf())```

### 💻 Minimal init:

```lua
vim.env.LAZY_STDPATH = ".repro"
load(vim.fn.system("curl -s https://raw.githubusercontent.com/folke/lazy.nvim/main/bootstrap.lua"))()

require("lazy.minit").repro({
    spec = {
        {
            "nvim-treesitter/nvim-treesitter",
            build = ":TSUpdate",

            ensure_installed = { "html", "latex", "markdown", "markdown_inline", "typst", "yaml" }
        },
        {
            "OXY2DEV/markview.nvim",
            -- branch = "dev",
            lazy = false
        }
    },
});
```

### 📜 Logs:

LOG | markview/autocmds.lua => bufHandle() -> array/with.md | Buffer state changed.
LOG |                                         markview/actions.lua => attach()                                          | Attached to 3.
LOG |                                  markview/actions.lua => autocmd() -> on_attach                                   |   Fired on_attach & MarkviewAttach.
LOG |                                              renderer.lua => clear()                                              |   Clearing: 3
LOG |                                              renderer.lua => clear()                                              |     Clearing total(3): 0ms
LOG |                                  markview/actions.lua => autocmd() -> on_disable                                  |   Fired on_disable & MarkviewDisable.

### Neovim version:

0.11.5

### ✅ Expected behavior:

All values ​​of ```:lua =require('markview.state').buf_attached(vim.api.nvim_get_current_buf())``` must be `true`.

**Comment by OXY2DEV** [comment]
See #456.
